org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacBindingResolver.java

```diff
@@ -271,6 +271,42 @@ private JavacPackageBinding preferentiallyInsertPackageBinding(JavacPackageBindi
 		public JavacTypeBinding getTypeBinding(JCTree tree, com.sun.tools.javac.code.Type type) {
 			return getTypeBinding(type, tree instanceof JCClassDecl);
 		}
+		
+		public IBinding getFieldAccessBinding(JCFieldAccess fieldAccess) {
```

Comment:
I don't think this should belong to the `Bindings` class, which is more a registry than a resolver. IMO, this method should be private in the host JavacBindingResolver class